### PR TITLE
@types/qunit - Add types for test.each for qunit

### PR DESCRIPTION
**Pull request by vstefanovic97**
Please fill in this template.

- [+] Use a meaningful title for the pull request. Include the name of the package modified.
- [+] Test the change in your own code. (Compile and run.)
- [+] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [+] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [+] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [+] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:

- [+] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [+] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

Basically this PR adds types for `test.each` which seems to be missing.
The method is documented here https://qunitjs.com/api/QUnit/test.each but it's missing from the types

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/qunit/index.d.ts
1. types/qunit/test/module-test.ts
1. types/qunit/test/global-test.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d7967cdde9880a6b742157298777f63e3f7283c0
</p>

**Review by gitKrystan — APPROVED** [review]
Lovely. Thank you.

File: types/qunit/index.d.ts
Comment (gitKrystan) [inline types/qunit/index.d.ts:562]: I approved but if we can add `only` and `todo` for completeness that would be 🔥

Comment (vstefanovic97) [reply types/qunit/index.d.ts:594]: @gitKrystan I added types for:
```typescript
test.skip
test.todo
test.only
test.if

test.each
test.todo.each
test.only.each
test.if.each
test.skip.each
```

**Comment by vstefanovic97** [comment]
@gitKrystan I think it's ready now

**Review by gitKrystan — APPROVED** [review]

**Comment by gitKrystan** [comment]
NICE. Thank you so much!

**Comment by gitKrystan** [comment]
Ready to merge

**Comment by boris-petrov** [comment]
Not sure if it matters that much but this commit broke somehow the overload resolution of the `test` function. That is, without it, `test` is typed as:

```
test<TestContext>(name: string, callback: (this: TestContext, assert: Assert) => void | Promise<unknown>): void (+1 overload)
```

With the new version of `@types/qunit`, it's typed as:

```
test(name: string, callback: QUnit.TestFunctionCallback): void
```

And I have to manually add a type for `this`.

**Comment by vstefanovic97** [comment]
@boris

> Not sure if it matters that much but this commit broke somehow the overload resolution of the `test` function. That is, without it, `test` is typed as:
> 
> ```
> test<TestContext>(name: string, callback: (this: TestContext, assert: Assert) => void | Promise<unknown>): void (+1 overload)
> ```
> 
> With the new version of `@types/qunit`, it's typed as:
> 
> ```
> test(name: string, callback: QUnit.TestFunctionCallback): void
> ```
> 
> And I have to manually add a type for `this`.

@boris-petrov looks like the overload comes from `ember-qunit`, I'll check it out and see if we need to fix this here or inside `ember-qunit`

**Comment by vstefanovic97** [comment]
@boris-petrov I created https://github.com/DefinitelyTyped/DefinitelyTyped/pull/70977 after this is merged I can create a PR to ember-qunit which will fix the overload

**Comment by boris-petrov** [comment]
@vstefanovic97 thank you for the quick response and fix!

**Pull request merged**
